js/app.js

```diff
@@ -0,0 +1,370 @@
+      this._updateHash();
+      this._map.panTo(this.getUserPosition());
+    }.bind(this));
+
+    this._map.addListener('zoom_changed', function() {
+      this._updateHash();
+    }.bind(this))
+  }
+
+  return this;
+};
+
+/**
+ * Updates the location hash to reflectthe user's marker's position and map
+ * zoom level.
+ * @private
+ * @return {this} TacoMap
+ */
+TacoMap.prototype._updateHash = function() {
+  var newPos = this.getUserPosition();
+  history.replaceState(null, null, '#' + newPos.lat + '_' + newPos.lng + '_' +
+      this._map.getZoom());
+  return this;
+};
+
+/**
+ * Adds a marker to the map's MarkerClusterer manager at the provided
+ * coordinates.
+ * @method
+ * @param {Object} coord - Coordinate object.
+ * @param {Object} coord.lat - Coordinate object's latitude.
+ * @param {Object} coord.lng - Coordinate object's latitude.
+ * @return {this} TacoMap
+ */
+TacoMap.prototype.addMarker = function(coord) {
+  if (!TacoMap.isValidCoord(coord)) {
+    return this;
+  }
+
+  var marker = new google.maps.Marker({
+    icon: {
+      anchor: new google.maps.Point(CONFIG.TACO_MAP.staticMarker.anchor.x,
+          CONFIG.TACO_MAP.staticMarker.anchor.y),
+      origin: new google.maps.Point(CONFIG.TACO_MAP.staticMarker.origin.x,
+          CONFIG.TACO_MAP.staticMarker.origin.y),
+      size: new google.maps.Size(CONFIG.TACO_MAP.staticMarker.height,
+          CONFIG.TACO_MAP.staticMarker.width),
+      url: CONFIG.TACO_MAP.staticMarker.url
+    },
+    position: coord,
+    zIndex: 9
+  });
+  this._mc.addMarker(marker);
+  return this;
+};
+
+/**
+ * Moves the center of the map to the provided coordinate and places the
+ * user's marker there.
+ * @method
+ * @param {Object} coord - Coordinate object.
+ * @param {Object} coord.lat - Coordinate object's latitude.
+ * @param {Object} coord.lng - Coordinate object's latitude.
+ * @return {this} TacoMap
+ */
+TacoMap.prototype.setCenter = function(coord) {
+  if (!TacoMap.isValidCoord(coord)) {
+    return this;
+  }
+
+  this._map.setCenter(coord);
+  this._userMarker.setPosition(coord);
+  return this;
+};
+
+/**
+ * Saves a marker to the database at the current user marker position and adds
+ * a new marker at that location.
+ * @method
+ * @return {this} TacoMap
+ */
+TacoMap.prototype.saveMarker = function() {
+  // Bounce the userMarker to indicate that something is happening.
+  this._userMarker.setAnimation(google.maps.Animation.BOUNCE);
+
+  // Fire a "saveStart" event on the map to provide an event hook.
+  $(this._map).trigger('saveStart');
+
+  // Update the database. Once it's successful, the "child_added" event will
+  // fire and a new marker will be added to the map at which point we can
+  // stop the bounce animation
+  var key = this._db.ref().child('trucks').push().key;
+  var update = {};
+  update['/trucks/' + key] = this.getUserPosition();
+  this._db.ref().update(update).then(function() {
```

Comment:
Maybe wrap in try/catch so that the icon doesn't bounce indefinitely in case of server error?